lib/pwwka/transmitter.rb

```diff
@@ -43,6 +44,7 @@ def initialize
     # - :ignore (aka as send_message_safely)
     # - :raise
     # - :resque -- use Resque to try to send the message later
+    # - :sidekiq -- use Sidekiq to try to send the message later
```

Comment:
I am not convinced that adding an `on_error: :sidekiq` option is the right way to go here. At this level of abstraction I think we're only concerned with answering the question of how to deal with errors we encounter when publishing messages to Rabbit, and I think that generic behaviors of "ignore," "raise," or "enqueue a job" are reasonable. I'm wondering if the "resque" and "sidekiq" options are too specific. (Putting aside for a minute that we likely have to maintain `on_error: :resque` to avoid introducing breaking changes.)

The consequence of this level of specificity (`on_error: resque` and `on_error: :sidekiq`) is that we end up with nearly identical `when :resque` and `when :sidekiq` cases, and then we end up almost writing another case statement in `.send_message_async`. 

Maybe's Dave's suggestion of a `background_job_processor:` option...

Reply:
This is a great point - in this area of the code we really do only care if we queue a job on errors. Elsewhere we can configure how that actually works.

Reply:
Agreed.  Encapsulating that distinction into the job strategy configured in `Pwwka.configuration.async_job_klass` would be better.